code/modules/mob/living/simple_animal/simple_animal_vr.dm

```diff
@@ -44,21 +44,25 @@
 		var/obj/belly/B = belly
 		for(var/mob/living/M in B)
 			new_fullness += M.size_multiplier
-		new_fullness = round(new_fullness, 1) // Because intervals of 0.25 are going to make sprite artists cry.
+	new_fullness = round(new_fullness, 1) // Because intervals of 0.25 are going to make sprite artists cry.
 	vore_fullness = min(vore_capacity, new_fullness)
 
-/mob/living/simple_animal/update_icon()
-	. = ..() // Call sideways "parent" to decide state
+/mob/living/simple_animal/proc/update_vore_icon()
 	if(vore_active)
 		update_fullness()
 		if(!vore_fullness)
-			return
-		else if(icon_state == icon_living && (vore_icons & SA_ICON_LIVING))
-			icon_state = "[icon_state]-[vore_fullness]"
-		else if(icon_state == icon_dead && (vore_icons & SA_ICON_DEAD))
-			icon_state = "[icon_state]-[vore_fullness]"
-		else if(icon_state == icon_rest && (vore_icons & SA_ICON_REST))
-			icon_state = "[icon_state]-[vore_fullness]"
+			return 0
+		else if((stat == CONSCIOUS) && (!icon_rest || !resting || !incapacitated(INCAPACITATION_DISABLED)) && (vore_icons & SA_ICON_LIVING))
+			return "[icon_living]-[vore_fullness]"
+		else if(stat >= DEAD && (vore_icons & SA_ICON_DEAD))
+			return "[icon_dead]-[vore_fullness]"
+		else if(((stat == UNCONSCIOUS) || resting || incapacitated(INCAPACITATION_DISABLED) ) && icon_rest && (vore_icons & SA_ICON_REST))
+			return "[icon_rest]-[vore_fullness]"
+	return 0
+
+/mob/living/simple_animal/lay_down()
+	..()
+	update_icons()
```

Comment:
. = ..() for safety when doing sideways overrides, in case someone alters the procs to use return values at some point.

What's the purpose of this, though? Is it for zshadows? That's the only thing done by /mob/living/update_icons. The simple animal proc is update_icon(). If it's for zshadows, you should give it to Polaris since they also use multiz and would benefit, and should really put it in update_icon() with a comment explaining why.

Reply:
update_icon() doesn't preserve scaling on simpleanimals

Reply:
Why?

Reply:
The resize_vr.dm only has the thing for update_icons(). Seems the resting verb uses update_icon() and apparently does the same for borgs too. #3394